packages/lockfile-lint/src/validators/index.js

```diff
@@ -39,8 +46,23 @@ function ValidateHostManager ({path, type, validatorValues, validatorOptions}) {
   const parser = new ParseLockfile(options)
   const lockfile = parser.parseSync()
   const validator = new ValidateHost({packages: lockfile.object})
+  const validationResult = validator.validate(validatorValues, validatorOptions)
 
-  return validator.validate(validatorValues, validatorOptions)
+  // Check if some of the errors are for allowed URLs and filter those out
+  if (validatorOptions && validatorOptions.allowedUrls) {
+    const urlValidator = new ValidateUrl({packages: lockfile.object})
+
+    validationResult.errors = validationResult.errors.filter(
```

Comment:
what happens if `validationResult.errors` is already populated from the host validator in line 49? (https://github.com/lirantal/lockfile-lint/pull/52/files#diff-b5633cfe76c3d56193e776b9dc02140dR49) we'll overwrite it here. It might be ok to overwrite *some* of the results because the url validation will allow them, but not others.

I think would be good to iterate specifically on each package there in the `errors` objects and then based on the results from `urlValidator.validateSingle()` into that instead of assigning to it and overwriting whatever is there. Since `validateSingle()` only returns true or false then you can't just pass the same to the `validationResult.errors` because that expects to have a specific structure, right? so I think that `validateSingle()` needs to return the same error object instead of a boolean. Either that, or you'll need to construct it properly here.

Reply:
@lirantal line 55 will filter out all errors that the did not pass the `HostValidator` check but passed the `UrlValidator` check. So, let's say you have `['npm', 'yarn']` specified for allowed hosts and some package fails the check because this packages resolves to a URL on Github. Now, we will have this error in the `validationResult.errors` array. However, if we explicitly specify this exact Github URL as allowed in our `allowedUrls` option then it will pass the `UrlValidator` check and will be filtered out. Is this not what we want?

Reply:
Yeah I think you're right. This check happens only for errors, and only for those that then have been sent through the allowed-urls so a failing check should keep the error object, and a passing one should remove it.